darwin/dataset/remote_dataset.py

```diff
@@ -366,35 +367,30 @@ def restore_archived(self, items) -> None:
 
     def fetch_annotation_type_id_for_name(self, name: str):
         """Fetches annotation type id for a annotation type name, such as bounding_box"""
-        annotation_types: Dict = self.client.get("/annotation_types")
-        for annotation_type in annotation_types:
+        if not self.annotation_types:
+            self.annotation_types = self.client.get("/annotation_types")
+        for annotation_type in self.annotation_types:
             if annotation_type["name"] == name:
                 return annotation_type["id"]
 
-    def create_annotation_class(self, name: str, type: str) -> Dict:
-        """
-        Creates an annotation class for this dataset.
-
-        Parameters
-        ----------
-        name : str
-            The name of the annotation class.
-        type : str
-            The type of the annotation class.
+    def create_annotation_class(self, name: str, type: str, subtypes: List[str] = []):
+        type_ids = []
+        for annotation_type in [type] + subtypes:
+            type_id = self.fetch_annotation_type_id_for_name(annotation_type)
+            if not type_id:
+                list_of_annotation_types = ", ".join([type["name"] for type in self.annotation_types])
```

Comment:
I have a doubt about this part. So, we are searching for `annotation_type` that may be `type` or something inside `subtypes`. 

Then, if we fail (`fetch_annotation_type_id_for_name` returns `None`) we are going to raise a ValueError saying that specific `annotation_type` was `Unknown` and we add `valid values: {list_of_annotation_types}"`. But if the only `valid values` are the ones inside `list_of_annotation_types` (computed from `self.annotation_types` could we just use the intersection between `subtypes` and `self.annotation_types`?

Reply:
But we also want to get all the type ids?

Reply:
After yesterday discussion, LGTM